types/precison_timestamp_types.go

```diff
@@ -0,0 +1,147 @@
+package types
+
+import (
+	"fmt"
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+// TimePrecision is used to represent the precision of a timestamp
+type TimePrecision int32
+
+const PrecisionUnknown TimePrecision = -1
+
+// precision values are proto values
+const (
+	PrecisionSeconds TimePrecision = iota
+	PrecisionDeciSeconds
+	PrecisionCentiSeconds
+	PrecisionMilliSeconds
+	PrecisionEMinus4Seconds // 10^-4 of seconds
+	PrecisionEMinus5Seconds // 10^-5 of seconds
+	PrecisionMicroSeconds
+	PrecisionEMinus7Seconds // 10^-7 of seconds
+	PrecisionEMinus8Seconds // 10^-8 of seconds
+	PrecisionNanoSeconds
+)
+
+func timePrecisionToProtoVal(val TimePrecision) int32 {
+	return int32(val)
+}
+
+func ProtoToTimePrecision(val int32) (TimePrecision, error) {
+	if val < 0 || val > 9 {
```

Comment:
use the constants here

`if val < PrecisionSeconds || val > PrecisionNanoSeconds {`